spec/interaction-spec.js

```diff
@@ -108,4 +195,711 @@ describe('c3 chart interaction', function () {
+
+                moveMouse(35, 268);
+
+                expect(document.querySelector('.c3-tooltip-container').style.display).toEqual('block');
+
+                const tooltipData = [...document.querySelectorAll('.c3-tooltip tr')];
+
+                expect(tooltipData.length).toBe(2); // header + data2
+
+                expect(tooltipData[1].querySelector('.name').textContent).toBe('data2');
+                expect(tooltipData[1].querySelector('.value').textContent).toBe('130');
+            });
+
+            it('expands only hovered bar', () => {
+                moveMouse(20, 20);
+
+                const barList = d3.selectAll('.c3-bar');
+
+                expect(barList.size()).toBeGreaterThan(0);
+
+                // nothing expanded
+                barList.each(function() {
+                    expect(this.classList.contains('_expanded_')).toBeFalsy();
+                });
+
+                moveMouse(38, 258);
+
+                barList.each(function() {
+                    if (this.classList.contains('c3-bar-0') && this.parentElement.classList.contains('c3-bars-data2')) {
+                        expect(this.classList.contains('_expanded_')).toBeTruthy();
+                    } else {
+                        expect(this.classList.contains('_expanded_')).toBeFalsy();
+                    }
+                });
+            });
+        });
+    });
+
+    describe('line chart', function() {
+        describe('tooltip_grouped=false', function() {
+            beforeAll(() => {
+                args = {
+                    data: {
+                        columns: [
+                            [ 'data1', 30, 200, 200, 400, 150, -250 ],
+                            [ 'data2', 130, -100, 100, 200, 150, 50 ],
+                            [ 'data3', 230, -200, 200, 0, 250, 250 ]
+                        ],
+                        type: 'line',
+                        groups: [
+                            [ 'data1', 'data2' ]
+                        ],
+                        onclick: function(d) {
+                            window.clickedData.push(d);
+                        }
+                    },
+                    tooltip: {
+                        grouped: false
+                    },
+                    axis: {
+                        x: {
+                            type: 'category'
+                        }
+                    },
+                    interaction: {
+                        enabled: true
+                    },
+                    point: {
+                        r: 2,
+                        sensitivity: 10,
+                        focus: {
+                            expand: {
+                                enabled: true,
+                                r: 8
+                            }
+                        }
+                    }
+                };
+            });
+
+            beforeEach(function() {
+                window.clickedData = [];
```

Comment:
I prefer putting this variable in more local scope as well (like `describe('line chart'`) to reduce the possibility of affecting the other tests in future.